### ISSUE TITLE
ERROR: logistic regression did not converge for phenotype X610.2. Perhaps increase --niter?

### ISSUE THREAD

**Issue by Shicheng-Guo** [rgcgithub/regenie#223]
Hi Joelle, 

I am trying to have a better understanding to the following error in 2.2.4

WARNING: Fitted probabilities numerically 0/1 occured (phenotype #X592.11).ERROR: logistic regression did not converge for phenotype X610.2. Perhaps increase --niter?

This error is the most frequent error in my large scale WGS and WES analysis. It occurs in some chromosome and phenotypes not all and will stop the analysis for the whole phenotype matrix (for example the overall phenotype=50, usually, I will split 3000 traits by 50 each file and input to regenie). I am wondering is there any way to predict how phenotype and chromosome will occur this issue, so that I can remove these phenotypes ahead. 

Thanks. 

Shicheng

```
Options in effect:
  --step 2 \
  --pred /home/sguo2/janssen4/regenie/2021Q4/binStep1/pred.bt.2021Q4.txt \
  --bed /expanse/projects/janssen4/ukbiobank_restricted/raw/WES/WES_450k_BED_BIM_FAM/ukb23148_c3.processed.vcf_processed \
  --keep /home/sguo2/janssen4/ukb/analytic/step1/qc_pass.id \
  --phenoFile /home/sguo2/data/PHENO/WES450K/UKB_PHECODE_Guo.vRegenie.IMP.2021Q4.1691.txt \
  --covarFile /home/sguo2/janssen4/ukb/WES-hg38/UKB.cov.r2.Guo.txt \
  --covarCol PC{1:20},Year,iSex,YearSquare \
  --catCovarList center \
  --maxCatLevels 22 \
  --anno-file /expanse/projects/janssen4/ukbiobank_restricted/pa_working/Annotation/UKB_QCed_450KWES_VEP/REGENIE_var_set/Canonical_Transcript/WES_chr3.anno \
  --set-list /expanse/projects/janssen4/ukbiobank_restricted/pa_working/Annotation/UKB_QCed_450KWES_VEP/REGENIE_var_set/Canonical_Transcript/WES_chr3.setlist \
  --mask-def /expanse/projects/janssen4/ukbiobank_restricted/pa_working/Annotation/UKB_QCed_450KWES_VEP/REGENIE_var_set/WES.masks \
  --aaf-bins 0.01,0.05 \
  --bt \
  --test additive \
  --spa \
  --approx \
  --pThresh 0.01 \
  --threads 12 \
  --bsize 1500 \
  --out /home/sguo2/janssen4/baklava/WES2021Q4Burden/UKB.phecode.v2021Q4.WES450.chr3.additive.burden

```

**Comment by joellembatchou** [comment]
Hi Shicheng,

This error usually occurs for BTs with low case counts (and even more often if there are many covariates). I have not done extensive analysis to build a single rule for when this occurs. In the next release, this will be fixed so that it skips the trait instead of stopping the run.

Cheers,
Joelle

**Comment by Shicheng-Guo** [comment]
Great. Thanks Joelle!! Looking forward to the new release in 2022!!

Shicheng

**Comment by joellembatchou** [comment]
Now implemented in v3.0.3
